pdfgen.c

```diff
@@ -3920,8 +3922,8 @@ static int parse_bmp_header(struct pdf_img_info *info, const uint8_t *data,
         snprintf(err_msg, err_msg_length, "BMP has negative width");
         return -EINVAL;
     }
-    if (info->bmp.biHeight == -2147483648) { // 1 << 31
-        snprintf(err_msg, err_msg_length, "BMP height overflow");
+     if (info->bmp.biHeight == 0x80000000) { // 1 << 31
```

Comment:
What was the warning that this prevented? Now it's a concern because `biHeight` is signed, but `0x80000000` is unsigned

Reply:
The original compare with -2147483648 generated on Visual Studio
"warning C4146: unary minus operator applied to unsigned type, result still unsigned"
with my change VS compiled without any warnings.

Reply:
Perhaps replacing the test with  
if (info->bmp.biHeight & 0x80000000) 
would work? it does on VS.

Reply:
That is just checking if it has a negative height. A negative height is legitimate - is is used to indicate that the image is flipped. It's a negative height that is too big to be inverted that is a problem (ie: that one specific value at the extreme negative end).

The other option would be to #include <limits.h> at the top, and do:
`if (info->bmp.biHeight == INT_MIN) {`

Reply:
I did a bit more research and it seems that the warning is triggered only on 32bit target (like x86) because -2147483648 cannot be represented as int in 32bits so it is promoted to unsigned int and then unary minus is applied. In limits.h, INT_MIN is defined as (-2147483647 - 1) and such construct do not cause the warning.  So, to get rid of the warning on 32bit targets the value -2147483648 used for comparizon should be changed to -2147483647 - 1. 
But personally I think that much better solution would be to use INT_MIN from limits.h as you suggested.

Reply:
Wow - I wasn't aware of that weird two-part construction of -2147483648. It makes sense, but still feels like it should work. Ok - let's go with the INT_MIN approach then?

Thanks for following up on this.

Reply:
Yes, INT_MIN is the best and most portable solution, will you make the change?